src/main/mainEvents.ts

```diff
@@ -13,75 +13,140 @@ interface MainEventNames {
-   'network-ready'() : void;
-   /**
-    * Emitted after the network has been disconnected or connected.
-    * @param online If true, then the network connection is (probably) working.
-    */
-   'update-network-status'(online: boolean): void;
-   /**
-    * Emitted when the integration state has changed.
-    *
-    * @param state A mapping of WSL distributions to the current state, or a
-    * string if there is an error.
-    */
-   'integration-update'(state: Record<string, boolean|string>): void;
-
-   /**
-    * Emitted as a request to get the credentials for API access.
-    */
-   'api-get-credentials'(): void;
-   /**
-    * Emitted as a reply to 'api-get-credentials'; the credentials can be used
-    * via HTTP basic auth on localhost.
-    *
-    * @note These credentials are meant for the UI; using them may require user
-    * interaction.
-    */
-   'api-credentials'(credentials: {user: string, password: string, port: number}): void;
+  'k8s-check-state'(mgr: VMBackend): void;
+
+  /**
+   * Fetch the currently stored settings.
+   * @note This may not match the currently active settings.
+   */
+  'settings-fetch'(): Settings;
+
+  /**
+   * Emitted when the settings have been changed.
+   *
+   * @param settings The new settings.
+   */
+  'settings-update'(settings: Settings): void;
+
+  /**
+   * Emitted to request that the settings be changed.
+   *
+   * @param settings The settings to change.
+   */
+  'settings-write'(settings: RecursivePartial<Settings>): void;
+
+  /**
+   * Emitted as a request to get the CA certificates.
+   */
+  'cert-get-ca-certificates'(): void;
+
+  /**
+   * Emitted as a reply to 'cert-get-ca-certificates'.
+   *
+   * @param certs The certificates found.
+   */
+  'cert-ca-certificates'(certs: (string | Buffer)[]): void;
+
+  /**
+   * Emitted after the network setup is complete.
+   */
+  'network-ready'(): void;
+
+  /**
+   * Emitted after the network has been disconnected or connected.
+   * @param online If true, then the network connection is (probably) working.
+   */
+  'update-network-status'(online: boolean): void;
+
+  /**
+   * Emitted when the integration state has changed.
+   *
+   * @param state A mapping of WSL distributions to the current state, or a
+   * string if there is an error.
+   */
+  'integration-update'(state: Record<string, boolean | string>): void;
+
+  /**
+   * Emitted as a request to get the credentials for API access.
+   */
+  'api-get-credentials'(): void;
+
+  /**
+   * Emitted as a reply to 'api-get-credentials'; the credentials can be used
+   * via HTTP basic auth on localhost.
+   *
+   * @note These credentials are meant for the UI; using them may require user
+   * interaction.
+   */
+  'api-credentials'(credentials: { user: string, password: string, port: number }): void;
 }
 
+type HandlerParams<eventName extends keyof MainEventNames> =
+  void extends ReturnType<MainEventNames[eventName]> ? never :
+  Parameters<MainEventNames[eventName]>;
```

Comment:
What does `Parameters<MainEventNames[eventName]>` resolve to?

Reply:
I will add comments; to clarify, though, this gives you the parameter types of the method (because `eventName` names some method).  [`Parameters<T>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype) is a TypeScript builtin.